### PR TITLE
Add bit LT embed to make unpack safe for any size

### LINKED ISSUE #899: Allow any size as an output of `unpack`
## Description

The unpack embed should accept any size of output

## Environment

- Compiler version: develop

## Steps to Reproduce

```
bool[256] a_bits = unpack::<256>(a)
```

This leads to `assertion failed: N <= 254`

But we should just pad with zeroes on the left. How to do that is not exactly clear, as the pad size is `N - 254` which would require making sure no underflow happens because of u32 semantics

Alternatively, this could be done in the `unpack` embed

### PR DESCRIPTION
**Pull request by Schaeff**
closes #899 

As a side effect, refactored code detecting constant expressions

### CODE REVIEW

File: zokrates_cli/examples/compile_errors/variable_constant_lt.zok [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from "EMBED" import bit_array_le
+
+// Unpack a field element as N big endian bits
```
Comment (dark64) [inline zokrates_cli/examples/compile_errors/variable_constant_lt.zok]: This comment seems outdated

File: zokrates_stdlib/stdlib/utils/pack/bool/unpack.zok [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
 // Unpack a field element as N big endian bits
 def main<N>(field i) -> bool[N]:
-
-	assert(N <= 254)
     
-    bool[N] res = unpack(i)
+    bool[N] res = unpack_unchecked(i)
+
+    assert(if N >= FIELD_SIZE_IN_BITS then bit_array_le(res, [...[false; N - FIELD_SIZE_IN_BITS], ...unpack_unchecked::<FIELD_SIZE_IN_BITS>(-1)]) else true fi)
```
Comment (dark64) [inline zokrates_stdlib/stdlib/utils/pack/bool/unpack.zok:10]: 💯

**Review by dark64 — COMMENTED** [review]
Nice one, some conflicting files atm but looks good to me!

**Comment by Schaeff** [comment]
@dark64 can you have another look?

File: zokrates_core/src/typed_absy/mod.rs [hunk 3]
```diff
@@ -1157,21 +1157,39 @@ pub struct StructExpression<'ast, T> {
...
+        &self.inner
+    }
+
+    pub fn as_inner_mut(&mut self) -> &mut StructExpressionInner<'ast, T> {
+        &mut self.inner
+    }
+
+    pub fn into_inner(self) -> StructExpressionInner<'ast, T> {
+        self.inner
+        // >>>>>>> 5a02186fc1d5c8f438a9663112f444497e752ea6
```
Comment (dark64) [inline zokrates_core/src/typed_absy/mod.rs]: and here

File: zokrates_core/src/typed_absy/mod.rs [hunk 4]
```diff
@@ -1157,21 +1157,39 @@ pub struct StructExpression<'ast, T> {
...
-            TypedExpression::Struct(e) => {
-                if e.ty() == &target_struct_ty {
-                    Ok(e)
-                } else {
-                    Err(TypedExpression::Struct(e))
-                }
-            }
-            e => Err(e),
-        }
+// <<<<<<< HEAD
```
Comment (dark64) [inline zokrates_core/src/typed_absy/mod.rs]: quick q, is this left on purpose or?

Comment (Schaeff) [reply zokrates_core/src/typed_absy/mod.rs]: oops

**Review by dark64 — APPROVED** [review]

**Pull request merged**